src/DotNetty.Transport/Channels/DefaultChannelId.cs

```diff
@@ -0,0 +1,419 @@
+﻿using DotNetty.Buffers;
+using DotNetty.Common.Utilities;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Globalization;
+using System.Linq;
+using System.Net;
+using System.Net.NetworkInformation;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Threading;
+
+namespace DotNetty.Transport.Channels
+{
+    [Serializable]
+    public class DefaultChannelId : IChannelId
+    {
+        // static final InternalLogger logger = InternalLoggerFactory.getInstance(DefaultChannelId.class);
+        static readonly Regex MachineIdPattern = new Regex("^(?:[0-9a-fA-F][:-]?){6,8}$");
+        const int MachineIdLen = 8;
+        static readonly byte[] MachineId;
+        const int ProcessIdLen = 4;
+        // Maximal value for 64bit systems is 2^22.  See man 5 proc.
+        // See https://github.com/netty/netty/issues/2706
+        const int MaxProcessId = 4194304;
+        static readonly int ProcessId;
+        const int SequenceLen = 4;
+        const int TimestampLen = 8;
+        const int RandomLen = 4;
+
+        static int nextSequence = 0;
+
+        static int seed = (int)(Stopwatch.GetTimestamp() & 0xFFFFFFFF); //used to safly cast long to int, because the timestamp returned is long and it doesn't fit into an int
+        static readonly ThreadLocal<Random> ThreadLocalRandom = new ThreadLocal<Random>(() => new Random(Interlocked.Increment(ref seed))); //used to simulate java ThreadLocalRandom
+
+        readonly byte[] data = new byte[MachineIdLen + ProcessIdLen + SequenceLen + TimestampLen + RandomLen];
+        int hashCode;
+
+        [NonSerialized]
+        string shortValue;
+
+        [NonSerialized]
+        string longValue;
+
+        static DefaultChannelId()
+        {
+            int processId = -1;
+            string customProcessId = SystemPropertyUtil.Get("io.netty.processId");
+            if (customProcessId != null)
+            {
+                int.TryParse(customProcessId, out processId);
```

Comment:
if `TryParse` fails, processId will be set to `0`. Please do `if (!int.TryParse(...) { processId = -1; }`